website/docs/reference/appsmith-framework/widget-actions/post-message.md

```diff
@@ -0,0 +1,73 @@
+# Post Message
+
+The **Post message** action uses the `postWindowMessage()` function to enable safe cross-origin communication between different [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) objects; for example, between an app and an iframe embedded within it. This may be useful in situations like:
+
+- Using Appsmith's [Iframe](/reference/widgets/iframe/) widget to embed other applications or pages.
+- Embedding Appsmith in an iframe of a parent application.
+
+## Signature
+
+```javascript
+postWindowMessage(message, targetIframe, targetOrigin)
+```
+
+### Arguments
+
+| Argument Name | Description |
+| ------------- | ----------- |
+| **message** | This is the message to send to the target iframe or window. Most JavaScript values are acceptable here, except `null` and `undefined`. (Default: `""`) |
+| **targetIframe** | This is the window to which you want to send the message. If its value is `"window"`, you are sending a message to the parent application’s window (where Appsmith is embedded). If you want to send a message to an iframe within Appsmith, enter the name of the iframe here. (Default: `"window"`) |
+| **targetOrigin** | This is the URL to which you can send messages. The default value of "`*`" means the message can be sent to any URL. If you want to limit sending messages to only the parent application (in which Appsmith is embedded), enter the URL of the parent application here. (Default: `"*"`) |
+
+To see examples of the **Post message** action, take a look at the [sample app](https://app.appsmith.com/applications/61f3d1949d6d6a6720c98681/pages/61f3d1949d6d6a6720c98684).
+
```

Comment:
ok here I recall earlier we called the sub-heading 'Examples', how about we call it 'Types of communication' or 'Levels of communication' and create an image like this one by Rimil (in the doc image, we wont include the WIP content, only the ones that are available today). I think it will help users understand very clearly before diving into each type

<img width="892" alt="Screenshot 2022-12-16 at 17 52 51" src="https://user-images.githubusercontent.com/32227135/208098054-6c87ddbe-ec73-403b-a422-7f3ebbe2633f.png">

We will have to update this image later when the other two communication levels also become available.

Reply:
great idea, these diagrams were extremely helpful

Reply:
image shared by Rimil
![Untitled-2022-11-16-0828](https://user-images.githubusercontent.com/32227135/209400859-0d73904e-e4a6-4387-9178-62ae59bfd3c4.png)